x-pack/plugins/siem/server/utils/build_validation/route_validation.test.ts

```diff
@@ -22,18 +46,41 @@ describe('buildRouteValidation', () => {
   });
 
   test('return validation error', () => {
-    const input = { id: 'someId' };
+    const input: Omit<Schema, 'ids'> & { id: string } = { id: 'someId' };
     const result = buildRouteValidation(schema)(input, validationResult);
 
-    expect(result).toEqual(
-      'Invalid value undefined supplied to : { ids: Array<string> }/ids: Array<string>'
-    );
+    expect(result).toEqual('Invalid value "undefined" supplied to "ids"');
   });
 
   test('return validated input', () => {
-    const input = { ids: ['someId'] };
+    const input: Schema = { ids: ['someId'] };
     const result = buildRouteValidation(schema)(input, validationResult);
 
     expect(result).toEqual(input);
   });
+
+  test('returns validation error if given extra keys on input for an array', () => {
```

Comment:
NOTE: These are the tests where it would fail without utilizing the exact check and allow extra keys to be sent in.